src/bin/dns/main.rs

```diff
@@ -49,8 +58,46 @@ fn main() {
     )
     .get_matches();
 
+  let mut module_resolvers: Vec<Box<ModuleResolver>> = std::vec::Vec::new();
```

Comment:
This is a lot of code that we'd need to duplicate in the other binaries. Maybe we need a shared rust module for those builders that all the binaries can use.